framework/utils/cached_ndarray.py

```diff
@@ -339,43 +339,73 @@ def append(self,entry):
     if type(entry) not in [np.ndarray]:
       raise IOError('Tried to add new data to cNDarray.  Can only accept np.ndarray, but got '+type(entry).__name__)
     # for now require full correct shape, later handle the single entry case
-    if len(entry.shape)!=2:
+    if len(entry.shape)!=1:
       # TODO single entry case
-      raise IOError('Tried to add new data to cNDarray.  Need shape (#,{}) but got "{}"!'.format(self.width,entry.shape))
+      raise IOError('Tried to add new data to cNDarray.  Need shape ({},) but got "{}"!'.format(self.width,entry.shape))
     # must have matching width (fix for single entry case)
-    if entry.shape[1] != self.width:
-      raise IOError('Tried to add new data to cNDarray.  Need {} entities per entry, but got '.format(self.width)+str(entry.shape[1]))
+    if entry.shape[0] != self.width:
+      raise IOError('Tried to add new data to cNDarray.  Need {} entries in array, but got '.format(self.width)+str(entry.shape[0]))
+    # check for index alignment, if it's not the first entry
+    if False:
```

Comment:
Why is it here?

Reply:
It was part of a benchmarking study.  I removed it.